public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -1,18 +1,190 @@
+  const evalNumber = safeParseDurationstr(currentEvaluation);
+  const forNumber = safeParseDurationstr(forDuration);
+  if (evalNumber === 0) {
+    return 0;
+  } else {
+    const evaluationsBeforeCeil = forNumber / evalNumber;
+    return evaluationsBeforeCeil < 1 ? 0 : Math.ceil(forNumber / evalNumber) + 1;
+  }
+};
+
+export const isRulerGrafanaRuleDTO = (rule: RulerRuleDTO): rule is RulerGrafanaRuleDTO => 'grafana_alert' in rule;
+export const isAlertingRuleDTO = (rule: RulerRuleDTO): rule is RulerAlertingRuleDTO => 'alert' in rule;
+
+export const getAlertInfo = (alert: RulerRuleDTO, currentEvaluation: string): AlertInfo => {
+  const emptyAlert: AlertInfo = {
+    alertName: '',
+    forDuration: '0s',
+    numberEvaluations: 0,
+  };
+  if (isRulerGrafanaRuleDTO(alert)) {
+    return {
+      alertName: alert.grafana_alert.title,
+      forDuration: alert.for,
+      numberEvaluations: getNumberEvaluationsToStartAlerting(alert.for, currentEvaluation),
+    };
+  }
+  if (isAlertingRuleDTO(alert)) {
+    return {
+      alertName: alert.alert,
+      forDuration: alert.for ?? '1m',
+      numberEvaluations: getNumberEvaluationsToStartAlerting(alert.for ?? '1m', currentEvaluation),
+    };
+  }
+  return emptyAlert;
+};
+export const isValidEvaluation = (evaluation: string) => {
+  try {
+    const duration = parsePrometheusDuration(evaluation);
+
+    if (duration < MIN_TIME_RANGE_STEP_S * 1000) {
+      return false;
+    }
+
+    if (duration % (MIN_TIME_RANGE_STEP_S * 1000) !== 0) {
+      return false;
+    }
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+export const getIntervalForGroup = (
+  rulerRules: RulerRulesConfigDTO | null | undefined,
+  group: string,
+  folder: string
+) => {
+  const folderObj: Array<RulerRuleGroupDTO<RulerRuleDTO>> = rulerRules ? rulerRules[folder] : [];
+  const groupObj = folderObj?.find((rule) => rule.name === group);
+
+  const interval = groupObj?.interval ?? MINUTE;
+  return interval;
+};
+
+export const safeParseDurationstr = (duration: string): number => {
+  try {
+    return parsePrometheusDuration(duration);
+  } catch (e) {
+    return 0;
+  }
+};
+
+type AlertsWithForTableColumnProps = DynamicTableColumnProps<AlertInfo>;
+type AlertsWithForTableProps = DynamicTableItemProps<AlertInfo>;
+
+export const RulesForGroupTable = ({
+  rulerRules,
+  group,
+  folder,
+}: {
+  rulerRules: RulerRulesConfigDTO | null | undefined;
+  group: string;
+  folder: string;
+}) => {
+  const styles = useStyles2(getStyles);
+  const folderObj: Array<RulerRuleGroupDTO<RulerRuleDTO>> = rulerRules ? rulerRules[folder] : [];
+  const groupObj = folderObj?.find((rule) => rule.name === group);
+  const rules: RulerRuleDTO[] = groupObj?.rules ?? [];
+
+  const { watch } = useFormContext<FormValues>();
+  const currentInterval = watch('groupInterval');
+
+  const rows: AlertsWithForTableProps[] = rules
+    .slice()
+    .sort(
```

Comment:
Could we first `map` the rules using `getAlertInfo` and then sort? It looks like we do the same job twice (we're calling `getAlertInfo` 3 times for each rule)

Reply:
Now I remember why I didn't do that. Because in the map we use the index for the `id`, and then if we sort the array, this `id` it's not working.

Reply:
I will review though!

Reply:
It worked!, done ✅